DyLib.hpp

```diff
@@ -14,15 +14,20 @@
 #include <functional>
 #include <exception>
 #include <utility>
+#ifdef export_symbol
+#error DyLib: "export_symbol" is already defined
+#endif
```

Comment:
If the symbol is already defined, I would just ignore and keep the already defined version. Could be useful if devs want to override your definition.

Also, it conventional to name this symbol `[LIB_NAME]_API`, so you could use `DYLIB_API` for example.

Reply:
Fixed on 3a088f6